db/cassandra/src/main/java/org/commonjava/indy/cassandra/data/CassandraStoreUtil.java

```diff
@@ -0,0 +1,83 @@
+    public static final String PORT = "port";
+    public static final String USER = "user";
+    public static final String PASSWORD = "password";
+    public static final String PROXY_HOST = "proxyHost";
+    public static final String PROXY_PORT = "proxyPort";
+    public static final String PROXY_USER = "proxyUser";
+    public static final String PROXY_PASSWORD = "proxyPassword";
+    public static final String KEY_CERT_PEM = "keyCertPem";
+    public static final String KEY_PASSWORD = "keyPassword";
+    public static final String TIMEOUT_SECONDS = "timeoutSeconds";
+    public static final String CACHE_TIMEOUT_SECONDS = "cacheTimeoutSeconds";
+    public static final String METADATA_TIMEOUT_SECONDS = "metadataTimeoutSeconds";
+    public static final String NFC_TIMEOUT_SECONDS = "nfcTimeoutSeconds";
+    public static final String PREFETCH_RESCAN = "prefetchRescan";
+    public static final String PREFETCH_RESCAN_TIMESTAMP = "prefetchRescanTimestamp";
+    public static final String MAX_CONNECTIONS = "maxConnections";
+    public static final String SERVER_CERT_PEM = "serverCertPem";
+    public static final String PASS_THROUGH = "passThrough";
+    public static final String IGNORE_HOST_NAME_VERIFICATION = "ignoreHostnameVerification";
+
+    // the attributes of hosted repository
+    public static final String STORAGE = "storage";
+    public static final String READONLY = "readonly";
+    public static final String SNAPSHOT_TIMEOUT_SECONDS = "snapshotTimeoutSeconds";
+
+    // the attributes of group repository
+    public static final String CONSTITUENTS = "constituents";
+    public static final String PREPEND_CONSTITUENT = "prependConstituent";
+
+    public static String getSchemaCreateKeyspace( String keySpace, IndyStoreManagerConfig config )
+    {
+        return "CREATE KEYSPACE IF NOT EXISTS " + keySpace
+                        + " WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':" + config.getReplicationFactor() + "};";
+    }
+
+    public static String getSchemaCreateTableStore( String keySpace )
+    {
+        return "CREATE TABLE IF NOT EXISTS " + keySpace + "." + TABLE_STORE + " ("
+                        + "packagetype varchar,"
+                        + "storetype varchar,"
+                        + "name varchar,"
+                        + "description varchar,"
+                        + "disabled boolean,"
+                        + "pathstyle varchar,"
+                        + "transientmetadata map<text, text>,"
+                        + "metadata map<text, text>,"
+                        + "pathmaskpatterns set<text>,"
+                        + "createtime varchar,"
+                        + "disabletimeout int,"
+                        + "authoritativeindex boolean,"
+                        + "rescaninprogress boolean,"
+                        + "extras map<text, text>,"
+                        + "PRIMARY KEY (( packagetype, storetype ), name )"
```

Comment:
This will effectively create 9 partitions (`{npm, maven, generic-http}` X `{remote, hosted, group}`) as of the packageTypes we have now. I wonder if that's enough uniqueness?

Reply:
Sorry John, do you mean the repo with packageType, storeType and the name is not enough to make them uniqueness ?

Reply:
I was wondering about it. I'm not sure

Reply:
Updates for the partition key: 0ab19eb